samtranslator/model/api/api_generator.py

```diff
@@ -338,15 +338,18 @@ def _construct_api_domain(self, rest_api):
         record_set_group = None
         if self.domain.get("Route53") is not None:
             route53 = self.domain.get("Route53")
-            if route53.get("HostedZoneId") is None:
+            if route53.get("HostedZoneId") is None and route53.get("HostedZoneName") is None:
                 raise InvalidResourceException(
                     self.logical_id, "HostedZoneId is required to enable Route53 support on Custom Domains."
```

Comment:
nit: "HostedZoneId or HostedZoneName is required to enable Route53 support on Custom Domains."

Reply:
Added